modules/@apostrophecms/doc/index.js

```diff
@@ -1093,6 +1097,67 @@ module.exports = {
           }
         });
       },
+
+      // Iterate through the document's field, and execute the provided handlers
+      // for each array, object and relationship field type.
+      // The handlers take the `doc` and the current `field` values as arguments,
+      // letting you edit them if needed.
+      //
+      // Note that the provided array and object handlers must call the `recursiveFunc`
```

Comment:
Why is `recursiveFunc` necessary — can't you do that part automatically after invoking each handler? The only reason I can think of why you wouldn't do that is if one of your applications of this method does *not* want to recurse. In which case, explicitly returning `false` (not undefined) could be a way to signal that, similar to how `false` is used in some lodash functions.

Reply:
We do want to recurse, but since we want to give the possibility to call `recursiveFunc` with mutated param (`doc`, 2nd argument), as it's done in `prepareForStorage`, I think that handlers do need to provided it.